packages/jsxstyle/jsxstyle.tsx

```diff
@@ -1,21 +1,30 @@
 import * as React from 'react';
 import { componentStyles, getStyleCache } from 'jsxstyle-utils';
+import { StyleCache } from 'jsxstyle-utils/src/getStyleCache';
 
-export const cache = getStyleCache();
+export const cache: StyleCache = getStyleCache();
 
-function factory(displayName, defaultProps, tagName) {
-  tagName = tagName || 'div';
-  if (
-    process.env.NODE_ENV === 'development' &&
-    (typeof displayName !== 'string' || !displayName)
-  ) {
-    throw new Error(
-      'makeReactStyleComponentClass expects param 1 to be a valid displayName'
-    );
-  }
+export interface JsxstyleProps {
+  className?: string;
+  component?: string | React.ComponentClass | React.SFC;
+  mediaQueries?: { [key: string]: string };
+  props?: { [key: string]: any };
+  style?: React.CSSProperties;
+}
+
+function factory(
+  displayName: string,
+  defaultProps?: { [key: string]: React.ReactText } | null
+) {
+  const tagName = 'div';
+
+  return class JsxstyleComponent extends React.Component<
+    JsxstyleProps & { [key: string]: any }
```

Comment:
I highly recommend removing `{[key: string]: any}` here, for the reason mentioned above.